tests/ui/redundant_test_prefix.rs

```diff
@@ -0,0 +1,11 @@
+#![warn(clippy::redundant_test_prefix)]
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_redundant_name() {
+        //
+    }
+}
```

Comment:
Could you add some test cases where the test functions are not in `#[cfg(test)]`, because it seems like this should only lint when the test `is_in_cfg_test`, but there are changes against it... well, got me a bit confused 🤔

Reply:
> it seems like this should only lint when the test is_in_cfg_test
Not quite, because test can also be defined outside of a a `#[cfg(test)] annotation (and typical "test" module). IIRC, there's at least one example of that in the diff from this PR.